### PR TITLE
Post Editor: fix default values of discussion options when editing new post

### PR DESCRIPTION
**Pull request by jsnajdr**
This PR adds a few more site options to request when fetching the sites list. This fixes a bug when editing a new post in the editor.

**First commit: Extract lists of site fields and options to a module with constants**
This patch deduplicates the list of site fields and options to fetch into `state/sites/constants` module. Until now, they were duplicated in state/jetpack-connect and state/sites and slowly getting out of sync. For example, `site_goals` was added only to one place in #22171.

Also, the list is now an array with one item per line. Makes for a much more readable diffs when a field is added or removed.

**Second commit: start fetching options for default values of post comments/pingback settings**
Request the `default_ping_status` and `default_comment_status` options when requesting sites: they are used as defaults when editing a new post.

**How to test:**
First, go to the Discussion settings of your site and edit the default values for "Allow link notifications from other blogs" and "Allow people to post comments": 
post-discussion-settings

Then, start editing a new blog post on that site and verify in the sidebar UI that the defaults are really applied:
post-discussion-good

Before this PR, the defaults were always empty:
post-discussion-bad
And were updated to the correct values only after first autosave. Which is triggered only after you enter a post title and start typing for a while.

### CODE REVIEW

File: client/state/sites/constants.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { join } from 'lodash';
+
+export const SITE_REQUEST_FIELDS = join( [
+	'ID',
```
Comment (blowery) [inline client/state/sites/constants.js]: nice, so much more readable! Can we also sort them a-z so it's easier to spot dupes?

File: client/state/sites/constants.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { join } from 'lodash';
+
+export const SITE_REQUEST_FIELDS = join( [
```
Comment (blowery) [inline client/state/sites/constants.js]: is there an advantage to lodash's join over the native `Array.prototype.join`?

Comment (jsnajdr) [reply client/state/sites/constants.js]: Not really. At first I thought that Lodash `join` has comma (`,`) as default separator char and `Array.prototype.join` doesn't, but it's not true -- it's the comma in both cases. I'll remove the Lodash import and call.

**Review by samouri — APPROVED** [review]
Very nice changes. I tested according to your instructions and everything checked out.

My only remaining question before merge is whether or not the additional field impacts performance of the API

File: client/state/sites/constants.js (hunk 1 above)
Comment (jsnajdr) [reply client/state/sites/constants.js]: All fields sorted and I also replaced Lodash with the native `Array.prototype.join`!

**Comment by jsnajdr** [comment]
> My only remaining question before merge is whether or not the additional field impacts performance of the API

Both options are simple boolean site options. They don't require any computation or extra SQL queries.

**Pull request merged**